jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/S3VirtualFields.kt

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package software.aws.toolkits.jetbrains.services.s3
+
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.vfs.VirtualFileSystem
+import java.io.InputStream
+import java.io.OutputStream
+import java.time.Instant
+
+/**
+ * BaseS3VirtualFile is a base class to represent a virtual file
+ */
+
+abstract class BaseS3VirtualFile(
+    val fileSystem: S3VirtualFileSystem,
```

Comment:
could go all one line - we're good till 140 chars - doesn't matter what @millems says.